src/components/generic/InputNumberNoScrollWithUnit/InputNumberNoScrollWithUnit.js

```diff
@@ -28,11 +28,40 @@ const InnerInput = styled(InputNumberNoScroll)`
   padding: ${theme.spacing.xsmall};
 `
 
-const InputNumberNoScrollWithUnit = ({ unit, alignUnitsLeft = false, ...restOfProps }) => {
+const InputNumberNoScrollWithUnit = ({
+  unit = '',
+  alignUnitsLeft = false,
+  value = '',
+  onChange = () => {},
+  name = '',
+  ...restOfProps
+}) => {
+  const [inputValue, setInputValue] = useState(value || '')
+
+  const handleInputChange = (e) => {
+    let rawValue = e.target.value
+
+    rawValue = rawValue.replace(/[^0-9.]/g, '')
+
+    // Restrict to a valid decimal format: up to two decimal places
+    if (rawValue.includes('.')) {
+      const [integerPart, decimalPart] = rawValue.split('.')
+      if (decimalPart.length > 2) {
+        rawValue = `${integerPart}.${decimalPart.slice(0, 2)}`
```

Comment:
using this instead of `toFixed(2)` as it is not ideal for real-time formatting

Reply:
Oh, interesting!!! GTK. Is it the converting the string to a number and then back on top of `toFixed` or is it just `toFixed` on its own? (if you happen to remember or notice, I dont need an answer if you didnt keep track) Im surprised!